maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java

```diff
@@ -619,7 +748,13 @@ public final void parseVersion(String version) {
     }
 
     private static Item parseItem(boolean isDigit, String buf) {
-        if (isDigit) {
+        return parseItem(false, isDigit, buf);
+    }
+
+    private static Item parseItem(boolean isCombination, boolean isDigit, String buf) {
```

Comment:
What is "buf"? It's not a StringBuffer. Is there a more descriptive name for this variable?

Reply:
This is the previous name, I think try not to change this name in this change, it will cause trouble to code review.